### PR TITLE
Support for redacted fields in KotlinGenerator

### LINKED ISSUE #813: Support redacted fields in KotlinGenerator
Example:
```proto
message Taco {
  optional string recipe = 1 [(squareup.redacted) = true];
}
```

This should:
1. Add `redacted = true` field to the `WireField` annotation.
2. Replace the value of the field with `██` in `toString()` output.

### PR DESCRIPTION
**Pull request by Egorand**
Closes #813

### CODE REVIEW

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 1]
```diff
@@ -722,6 +757,78 @@ class KotlinGenerator private constructor(
...
+  private fun redactFun(message: MessageType): FunSpec {
+    val className = nameToKotlinName.getValue(message.type())
+    val nameAllocator = nameAllocator(message)
+    val result = FunSpec.builder("redact")
+        .addModifiers(OVERRIDE)
+        .addParameter("value", className)
+        .returns(className.copy(nullable = true))
+
+    val redacted = message.fields().filter { it.isRedacted }
+    val requiredRedacted = redacted.filter { it.isRequired }
```
Comment (oldergod) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: can we rename it so it conveys the object is actually a list?

File: wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/MessageUsingMultipleEnums.kt [hunk 2]
```diff
@@ -69,6 +69,11 @@ data class MessageUsingMultipleEnums(
           unknownFields = unknownFields
         )
       }
+
+      override fun redact(value: MessageUsingMultipleEnums): MessageUsingMultipleEnums? =
+          value.copy(
+        unknownFields = ByteString.EMPTY
+      )
```
Comment (oldergod) [inline wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/MessageUsingMultipleEnums.kt]: I don't like this indent, is it expected behavior or could this be a poet bug?

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 3]
```diff
@@ -444,6 +451,7 @@ class KotlinGenerator private constructor(
           .useSiteTarget(FIELD)
           .addMember("tag = %L", field.tag())
           .addMember("adapter = %S", field.getAdapterName(nameDelimiter = '#'))
+          .apply { if (field.isRedacted) addMember("redacted = true") }
```
Comment (oldergod) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: :sweet-kotlin:

File: wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/MessageUsingMultipleEnums.kt (hunk 2 above)
Comment (Egorand) [reply wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/MessageUsingMultipleEnums.kt]: The indentation for arguments is applied manually. It's a missing KP feature that I'm working on.

**Comment by Egorand** [comment]
@oldergod following up on our offline discussion: I'm gonna merge this as is since it at least follows the approach of the `JavaGenerator`, if we come up with a better solution in the future I'll be happy to submit a followup PR.

**Pull request merged**